File: app/src/main/java/com/oxodiceproductions/dockmaker/Database/ImageDao.java
package com.oxodiceproductions.dockmaker.Database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface ImageDao {
    @Query("select * from Image")
    List<Image> getAll();

    @Insert
    long insert(Image image);

    @Delete
    void delete(Image image);

   /* @Query("update Image set imageIndex=:newIndex where id=:id")
    void updateIndex(long id, int newIndex);*/

    @Query("delete from Image")
    void deleteAll();

    @Query("select * from Image where docId=:docId")
    List<Image> getImagesByDocId(long docId);

    @Query("delete from Image where imagePath=:imagePath")
    void deleteImageByPath(String imagePath);

    @Update
    void update(Image image);

    @Query("select * from Image where id=:id and imageIndex=:imageIndex")
    Image getImageByIndex(long id,int imageIndex);
}

File: app/src/main/java/com/oxodiceproductions/dockmaker/database/ImageDao.kt
package com.oxodiceproductions.dockmaker.database

import androidx.room.*

@Dao
interface ImageDao {
    @Query("select * from Image")
    suspend fun all(): List<Image?>?

    @Insert
    suspend fun insert(image: Image?): Long

    @Delete
    suspend fun delete(image: Image?)

    /* @Query("update Image set imageIndex=:newIndex where id=:id")
    void updateIndex(long id, int newIndex);*/
    @Query("delete from Image")
    suspend fun deleteAll()

    @Query("select * from Image where docId=:docId")
    suspend fun getImagesByDocId(docId: Long): List<Image?>

    @Query("delete from Image where imagePath=:imagePath")
    suspend fun deleteImageByPath(imagePath: String)

    @Update
    suspend fun update(image: Image)

    @Query("select * from Image where id=:id and imageIndex=:imageIndex")
    suspend fun getImageByIndex(id: Long, imageIndex: Int): Image

    @Query("select * from Image where docId=:docId and imagePath=:imagePath")
    suspend fun getImageByImagePath(docId: Long, imagePath: String?): Image
}